api/cluster.go

```diff
@@ -303,12 +304,24 @@ func (s *Server) localTagDelByQuery(request models.IndexTagDelByQuery) (int, err
 		return 0, err
 	}
 
-	if request.Execute {
+	if !request.Execute {
+		// Use FindTerms to count what would be deleted
+		matched, _ := s.MetricIndex.FindTerms(request.OrgId, []string{}, query)
+		return int(matched), nil
+	}
+
+	if request.Method == "archive" {
+		casidx, ok := s.MetricIndex.(*cassandra.CasIdx)
+		if !ok {
+			return 0, errors.New("Only cassandra index supports 'archive' method")
+		}
+		deleted, err := casidx.ArchiveTagged(request.OrgId, query)
+		return len(deleted), err
+	} else if request.Method == "delete" {
 		deleted, err := s.MetricIndex.DeleteTagged(request.OrgId, query)
 		return len(deleted), err
 	}
-	matched, _ := s.MetricIndex.FindTerms(request.OrgId, []string{}, query)
-	return int(matched), nil
+	return 0, fmt.Errorf("Unknown delete method '%s'", request.Method)
```

Comment:
can `request.Method` be any other operations, or even empty here? If that is the case, nothing will be deleted.

Reply:
It shouldn't happen because we have validation on [the input](https://github.com/bloomberg/metrictank/pull/105/files#diff-d86a4cc7a69f5834cefe5049137eea3980718bdc574d6dcbfb0a7eeded11e6ebR172) but this is defensive code. Not deleting anything and returning an error is the desired behavior.